src/ts/order.ts

```diff
@@ -59,6 +59,17 @@ export interface Order {
    * Specifies whether or not the order is partially fillable.
    */
   partiallyFillable: boolean;
+  /**
+   * Specifies how the sell token balance will be withdrawn. It can either be
+   * taken using ERC20 token allowances made directly to the Vault relayer
+   * (default) or using Balancer Vault internal or external balances.
+   */
+  sellTokenBalance?: OrderBalance;
+  /**
+   * Specifies how the buy token balance will be paid. It can either be paid
+   * directly in ERC20 tokens (default) in Balancer Vault internal balances.
+   */
+  buyTokenBalance?: OrderBalance;
```

Comment:
Why are the two flags optional, considering that all other flags are mandatory? What does the absence of them represent in the contract?
Also, how much complexity would it add to type them according to what their options are?

Reply:
Created #615